stdlib/public/core/CTypes.swift

```diff
@@ -100,6 +100,9 @@ public typealias CLongDouble = Double
 /// The C++ 'wchar_t' type.
 public typealias CWideChar = Unicode.Scalar
 
+/// The C++20 'char8_t' type, which has UTF-8 encoding.
+public typealias CChar8 = UInt8
```

Comment:
@airspeedswift, how are we handling availability for new top-level typealiases? These can't affect ABI.

Reply:
Also, I suggested Varun pick this up as a starter-ish task, but this is a public change to the stdlib, and `CChar8` is close to `CChar`. Does it need formal review or even a proposal?

Reply:
Hmm, given that WG14 is working on a TN for `char16_t` and `char32_t`, I could see `char8_t` being introduced into C, where it gets interesting with `-fsigned-char` and `-fno-signed-char`.  I think that on Darwin targets where you default to `-fsigned-char`, having `CChar` be `signed char`, and `CChar8` be `unsigned char` is slightly scary.  Might be worth bringing this up through a proposal.

Reply:
@compnerd but what's the alternative, really? Yeah, it would be slightly messy, but there's really no other option on the table except maybe "just don't import these types". I think this is a thing that we just live with.

Reply:
I'm not convinced that "just don't import these types" is a viable solution (particularly long term).  Sure, we can get away with it for the time being, but, I don't think that we would be able to escape it forever.

Now, I'm not saying that this is what we should do or even that what I'm suggesting is good.  Hopefully someone can come up with something more clever.

We could import this as `CXX.Char8` (where `CXX` is a `@frozen` enum which I believe should allow inling).

Reply:
For the record, non-generic containing contexts don't really affect codegen at all for nested types. So yeah, that is technically an option. It's a weird choice if C22 adopts `char8_t` as well, though.

Reply:
But we could introduce `CChar8` at that time for the C `char8_t` and if they make it compatible with `char` (including `-fsigned-char`), we can handle both.

Reply:
I can't see `char8_t` ever being signed; it's defined as a UTF-8 code unit. And we can't introduce `CChar8` separate from `CXX.Char8` because they'd be different types you could overload on (unless one is a typealias).

Reply:
Oops, they're all typealiases. Retracted, at least partially.